sdk/lib/rtl/path.c

```diff
@@ -1617,6 +1617,8 @@ RtlDetermineDosPathNameType_U(IN PCWSTR Path)
     /* Unlike the newer RtlDetermineDosPathNameType_U we assume 4 characters */
     if (IS_PATH_SEPARATOR(Path[0]))
     {
+        if ((Path[1] == L'?') && (Path[2] == L'?') &&
```

Comment:
This is wrong: this is not a DOS path, and it breaks compatibility with Windows.

Reply:
yep, I know. But ROS replaces \\?\ path with \??\ when creating symbolic links

Reply:
see deviface.c:1276      ->    SymbolicLinkName->Buffer[1] = '?';

Reply:
Then fix the place that needs to be fixed, not that function. Also there was a PR by @oleg-dubinskiy regarding those symbolic links; maybe it could help with the correct fix.

Reply:
unfortunately here is 240+ places where \??\ is hardcoded in the ROS. Plus some places where path is converted \\?\ -> \??\ and backwards.
It will be a big work to fix that

Reply:
`\\?\` is a Win32-only style of path. NTOS uses `\??\`, which is what's expected.
Also, you haven't explained at all why there are problems with this `\\?\` thing. So this PR is most likely to be rejected.

Reply:
I will test 3510 tomorrow. 
Problem is related to my another PRs, until I did this hack programs fail to open device from symlink.
I was asking about /??/ path but newer got any reaction. So I decided to post it here to discuss.

Anyway as long as I need to fix build, I will also remove this bit of code too. Just need to find real solution for this problem

Thank you<

Reply:
Hi, again!

So, 3510 does not affect on this issue. Problem in the CM_Get_Device_Interface_ListW using that function resulting InterfaceList contains /??/ path, that cannot be opened in user mode.

here is my testes:
![image](https://user-images.githubusercontent.com/52833910/189478866-5cca737a-6381-4fec-abb7-7cf314b9794d.png)
![image](https://user-images.githubusercontent.com/52833910/189479201-c49f795e-b98d-424a-afa9-da1ec0f7fea2.png)

So, to conclude. Looks like NtPlugPlayControl(PlugPlayControlGetInterfaceDeviceList) must have ability to control IoGetDeviceInterfaces //?/ -> /??/ conversion. To be able to not send kernel internal address into the user mode.

Reply:
@oleg-dubinskiy Do you have an idea what might be going on there? Since your PR #3510 also checks for those PnP symbolic links containing `\\?\` and `\??\` and seems to try to handle conversion between both?